rekolekcje-engine/src/verify/java/pl/oaza/warszawa/dor/rekolekcje/api/participants/storage/ParticipantsStorageExpectations.java

```diff
@@ -19,45 +23,30 @@ public ParticipantsStorageExpectations(ParticipantsDatabase database) {
     final List<Long> idsInSystem = database.getAllParticipantData().stream()
-        .map(ParticipantData::getId)
+        .map(ParticipantSampleData::getId)
         .collect(toList());
     assertThat(idsInSystem).doesNotContain(id);
   }
 
-  private void assertThatDataIsTheSame(ParticipantDTO dto, ParticipantData data) {
-    assertThat(dto.getFirstName()).isEqualTo(data.getFirstName());
-    assertThat(dto.getLastName()).isEqualTo(data.getLastName());
-    assertThat(dto.getPesel()).isEqualTo(data.getPesel());
-    assertThat(dto.getParishId()).isEqualTo(data.getParishId());
-    assertThat(dto.getPersonalData().getFatherName()).isEqualTo(data.getFatherName());
-    assertThat(dto.getPersonalData().getMotherName()).isEqualTo(data.getMotherName());
-    assertThat(dto.getPersonalData().getChristeningPlace()).isEqualTo(data.getChristeningPlace());
-    compareDates(dto.getPersonalData().getChristeningDate(), convertToUtc(data.getChristeningDate()));
-    assertThat(dto.getPersonalData().getEmergencyContactName()).isEqualTo(data.getCloseRelativeName());
-    assertThat(dto.getPersonalData().getEmergencyContactNumber()).isEqualTo(data.getCloseRelativeNumber());
-    assertThat(dto.getAddress().getStreetName()).isEqualTo(data.getStreet());
-    assertThat(dto.getAddress().getStreetNumber()).isEqualTo(data.getStreetNumber());
-    assertThat(dto.getAddress().getFlatNumber()).isEqualTo(data.getFlatNumber());
-    assertThat(dto.getAddress().getPostalCode()).isEqualTo(data.getPostalCode());
-    assertThat(dto.getAddress().getCity()).isEqualTo(data.getCity());
-    compareDates(dto.getExperience().getKwcSince(), convertToUtc(data.getKwcSince()));
-    assertThat(dto.getExperience().getKwcStatus()).isEqualTo(data.getKwcStatus());
-    assertThat(dto.getExperience().getNumberOfCommunionDays()).isEqualTo(data.getNumberOfCommunionDays());
-    assertThat(dto.getExperience().getNumberOfPrayerRetreats()).isEqualTo(data.getNumberOfPrayerRetreats());
+  public void correctDataIsPersisted(ParticipantDTO dto) {
+    final ParticipantSampleData persistedData = database.getPersistedData(dto);
+    assertThat(persistedData.getId()).isNotNull();
+    assertThat(persistedData.getFirstName()).isEqualTo(dto.getFirstName());
+    assertThat(persistedData.getLastName()).isEqualTo(dto.getLastName());
+    assertThat(persistedData.getPesel()).isEqualTo(dto.getPesel());
+    assertThat(persistedData.getParishId()).isEqualTo(dto.getParishId());
+    compareDates(dto.getPersonalData().getChristeningDate(), convertToUtc(persistedData.getChristeningDate()));
+    assertThat(persistedData.getPostalCode()).isEqualTo(dto.getAddress().getPostalCode());
+    assertThat(persistedData.getCurrentTreatment()).isEqualTo(dto.getHealthReport().getCurrentTreatment());
+    assertThat(persistedData.getKwcStatus()).isEqualTo(dto.getExperience().getKwcStatus());
+    assertThat(persistedData.getNumberOfCommunionDays()).isEqualTo(dto.getExperience().getNumberOfCommunionDays());
```

Comment:
`assertThat(...).isEqualToComparingFieldByFieldRecursively(...)`
Nie można zastosować tej metody?

Reply:
Nie zastosowałem tego, bo po pierwsze `ParticipantSampleData` nie musi się całkowicie pokrywać z DTO, i może się zdarzyć, że chcemy sprawdzić, czy dla w pełni wypełnionego DTO mamy zapisane poprawnie pola. I wtedy jeśli mamy taką metodę to `ParticipantSampleData` musi pewnie mieć takie same pola, czyli rozrasta nam się tak samo jak DTO. A cały myk polega na tym, że teraz mamy chudą klaskę, i sprawdziamy tylko po jednym polu każdego typu. 
Drugą kwestią jest data. DTO ma datę typu java.time, natomiast z bazy pobierana jest data w formacie z pakietu sql, dlatego trzeba dokonać konwersji, chociażby dokleić strefę czasową.